spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmountStyle.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.NumberUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@code DigitalAmount} format styles.
+ *
+ * @author Dmytro Nosan
+ * @since 2.1.0
+ */
+public enum DigitalAmountStyle {
```

Comment:
Please move the logic of that class in a simplified version of `parse` on DurationSize` itself. There is no need to expose such open option at this point.